mongo/integration/csot_prose_test.go

```diff
@@ -89,55 +89,83 @@ func TestCSOTProse(t *testing.T) {
 		mt.RunOpts("serverSelectionTimeoutMS honored if timeoutMS is not set", mtOpts, func(mt *mtest.T) {
 			mt.Parallel()
 
-			callback := func(ctx context.Context) {
-				err := mt.Client.Ping(ctx, nil)
+			callback := func() bool {
+				err := mt.Client.Ping(context.Background(), nil)
 				assert.NotNil(mt, err, "expected Ping error, got nil")
+				return true
 			}
 
 			// Assert that Ping fails within 150ms due to server selection timeout.
-			assert.Soon(mt, callback, 150*time.Millisecond)
+			assert.Eventually(t,
+				func() bool {
+					return callback()
+				},
+				150*time.Millisecond,
+				10*time.Millisecond,
+				"expected ping to fail within 150ms")
 		})
 
 		cliOpts = options.Client().ApplyURI("mongodb://invalid/?timeoutMS=100&serverSelectionTimeoutMS=200")
 		mtOpts = mtest.NewOptions().ClientOptions(cliOpts).CreateCollection(false)
 		mt.RunOpts("timeoutMS honored for server selection if it's lower than serverSelectionTimeoutMS", mtOpts, func(mt *mtest.T) {
 			mt.Parallel()
 
-			callback := func(ctx context.Context) {
-				err := mt.Client.Ping(ctx, nil)
+			callback := func() bool {
+				err := mt.Client.Ping(context.Background(), nil)
 				assert.NotNil(mt, err, "expected Ping error, got nil")
+				return true
 			}
 
 			// Assert that Ping fails within 150ms due to timeout.
-			assert.Soon(mt, callback, 150*time.Millisecond)
+			assert.Eventually(t,
+				func() bool {
+					return callback()
+				},
+				150*time.Millisecond,
+				10*time.Millisecond,
+				"expected ping to fail within 150ms")
 		})
 
 		cliOpts = options.Client().ApplyURI("mongodb://invalid/?timeoutMS=200&serverSelectionTimeoutMS=100")
 		mtOpts = mtest.NewOptions().ClientOptions(cliOpts).CreateCollection(false)
 		mt.RunOpts("serverSelectionTimeoutMS honored for server selection if it's lower than timeoutMS", mtOpts, func(mt *mtest.T) {
 			mt.Parallel()
 
-			callback := func(ctx context.Context) {
-				err := mt.Client.Ping(ctx, nil)
+			callback := func() bool {
+				err := mt.Client.Ping(context.Background(), nil)
 				assert.NotNil(mt, err, "expected Ping error, got nil")
+				return true
 			}
 
 			// Assert that Ping fails within 150ms due to server selection timeout.
-			assert.Soon(mt, callback, 150*time.Millisecond)
+			assert.Eventually(t,
+				func() bool {
+					return callback()
+				},
+				150*time.Millisecond,
+				20*time.Millisecond,
```

Comment:
Since the callback only returns true, we might as well set the tick to 1ms. This ensures the callback is called within 1ms of the test starting.